overlord/snapstate/snapstate_install_test.go

```diff
@@ -586,7 +586,6 @@ func (s *snapmgrTestSuite) TestInstallStrictIgnoresClassic(c *C) {
 	defer restore()
 
 	s.state.Lock()
-	defer s.state.Unlock()
```

Comment:
But then if an assertion fails the state will remain locked. I'm not sure if this is a problem, but it's something to double-check.

Other tests solve the issue by adding a `defer Lock()` instead: https://github.com/snapcore/snapd/blob/master/overlord/hookstate/ctlcmd/refresh_test.go#L288-L324

Reply:
> But then if an assertion fails the state will remain locked. I'm not sure if this is a problem, but it's something to double-check.

I don't think it's a problem because the state is recreated before each test.

Reply:
this doesn't really solve the problem if there's a panic out of some other function. Also it introduces code that is not recommended for production. As @mardy mentioned the only acceptable solution vs the status quo is using more:

``
state.Lock()
defer state.Unlock()

state.Unlock()
defer state.Lock()
```

combos.

One approach could be to put a `state.Unlock()/defer state.Lock()` at the start of settle itself. Not sure how many tests would be hard to tweak wrt that change though.

Reply:
> this doesn't really solve the problem if there's a panic out of some other function. 

I'm a bit confused by this part. The problem I was trying to solve was the panic caused by a double unlock when the settle assertion failed. What do you mean by a panic in another function?

Reply:
snapstate.Install  because of change panics, we leave the test now with the lock held, this might work or not, TearDown might not have issues because it could need to lock unlocked so we would get a different kind of double panic. Also as I said this new code is far from production patterns, which should be either:

* Lock/Unlock + defer of reverse
* Lock/Unlock or the reverse around a small amount of code.  This is now does neither of these